python/pyarrow/_compute.pyx

```diff
@@ -1091,6 +1091,22 @@ class AssumeTimezoneOptions(_AssumeTimezoneOptions):
         self._set_options(timezone, ambiguous, nonexistent)
 
 
+cdef class _WeekOptions(FunctionOptions):
+    def _set_options(self, week_starts_monday, count_from_zero,
+                     first_week_is_fully_in_year):
+        self.wrapped.reset(
+            new CWeekOptions(week_starts_monday, count_from_zero,
+                             first_week_is_fully_in_year)
+        )
+
+
+class WeekOptions(_WeekOptions):
+    def __init__(self, week_starts_monday=True, count_from_zero=False,
```

Comment:
These should be keyword-only arguments, as `WeekOptions(False, False, True)` is basically write-only code.

Reply:
Done.